curvefs/src/client/s3/client_s3_adaptor.h

```diff
@@ -0,0 +1,119 @@
+/*
+ *  Copyright (c) 2020 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: curve
+ * Created Date: 21-5-31
+ * Author: huyao
+ */
+#ifndef CURVEFS_SRC_CLIENT_S3_CLIENT_S3_ADAPTOR_H_
+#define CURVEFS_SRC_CLIENT_S3_CLIENT_S3_ADAPTOR_H_
+
+#include <vector>
+#include <string>
+
+#include "curvefs/proto/mds.pb.h"
+#include "curvefs/proto/metaserver.pb.h"
+#include "curvefs/proto/space.pb.h"
+#include "curvefs/proto/common.pb.h"
+#include "curvefs/src/client/s3/client_s3.h"
+
+namespace curvefs {
+namespace client {
+
+using curvefs::metaserver::S3ChunkInfo;
+using curvefs::metaserver::Inode;
+using curvefs::metaserver::S3ChunkInfoList;
+using curvefs::metaserver::UpdateInodeS3VersionRequest;
+using curvefs::metaserver::UpdateInodeS3VersionResponse;
+
+
+using curvefs::space::AllocateS3ChunkRequest;
+using curvefs::space::AllocateS3ChunkResponse;
+
+/*
+using namespace curvefs::metaserver;
+using namespace curvefs::space;
+*/
+
+struct S3ClientAdaptorOption {
+    uint64_t blockSize;
+    uint64_t chunkSize;
+    std::string metaServerEps;
+    std::string allocateServerEps;
+};
+
+struct S3ReadRequest {
+    S3ChunkInfo chunkInfo;
+    uint64_t readOffset;
+};
+
+struct S3ReadResponse {
+    uint64_t readOffset;
+    uint64_t bufLen;
+    char* dataBuf;
```

Comment:
`dataBuf` is leaked, and it's lefttime is not clear. I think it better to add constructor and destructor for `S3Response`, and new and delete in those functions. Also, you should carefully handle deep copy and shalow copy

Reply:
done